simtools/util/validate_data.py

```diff
@@ -96,6 +109,50 @@ def _check_required_columns(self):
                     self._logger.error("Missing required column '{}'".format(key))
                     raise
 
+    def _sort_data(self):
+        """
+        Sort data according to one data column
+        (if required by any column attribute)
+
+        Data is either sorted or reverse sorted
+        """
+
+        _columns_to_sort = []
+        _columns_to_reverse_sort = []
+        for key, value in self._reference_data_columns.items():
+            if 'attribute' in value:
+                if 'sort' in value['attribute']:
+                    _columns_to_sort.append(key)
+                elif 'reversesort' in value['attribute']:
+                    _columns_to_reverse_sort.append(key)
+
+        if len(_columns_to_sort) > 0:
+            self._logger.debug("Sorting data columns: {}".format(
+                _columns_to_sort))
+            self.data_table.sort(_columns_to_sort)
+        elif len(_columns_to_reverse_sort) > 0:
+            self._logger.debug("Reverse sorting data columns: {}".format(
+                _columns_to_reverse_sort))
+            self.data_table.sort(_columns_to_reverse_sort, reverse=True)
+
+    def _check_data_for_duplicates(self):
+        """
+        Remove duplicates from data columns
+        (if required by any column attribute)
+
+        Note: checks values in given column and removes
+        duplicates for complete data row (even of other
+        columns have different values). Expectation is that
+        there is one column only with column attributes)
+
+        """
+
+        for key, value in self._reference_data_columns.items():
+            if 'attribute' in value and 'no_dublicates' in value['attribute']:
+                self._logger.debug("Removing duplicates for column '{}'".format(
+                    key))
+                self.data_table = unique(self.data_table)
```

Comment:
I haven't tried it, so I am not sure about it, but I think that if the wavelength is the same, but the QE is different, this will not remove that row. I also am not sure if we want simply remove that row in that case. We should probably raise an error and stop in this case.
Also, in the case that you do find duplicate wavelengths, we should at least raise a warning so the user is aware and can make sure it is OK to simply remove it.

Reply:
Improved this function now:
- test that if e.g.,  we remove duplicated wavelength values, the corresponding data columns are the same. Otherwise raise a ValueError
- allow to test for uniqueness of several columns

Added also a corresponding test.

Commit a73681e